x-pack/test/api_integration/apis/ml/annotations/update_annotations.ts

```diff
@@ -0,0 +1,171 @@
+      bucket_span: '15m',
+      influencers: [],
+      detectors: [
+        {
+          function: 'mean',
+          field_name: 'responsetime',
+        },
+        {
+          function: 'sum',
+          field_name: 'responsetime',
+        },
+      ],
+    },
+    data_description: { time_field: '@timestamp' },
+    analysis_limits: { model_memory_limit: '10mb' },
+  }));
+  const jobIds = testSetupJobConfigs.map((j) => j.job_id);
+
+  const createAnnotationRequestBody = (jobId: string) => {
+    return {
+      timestamp: Date.now(),
+      end_timestamp: Date.now(),
+      annotation: 'Test annotation',
+      job_id: jobId,
+      type: ANNOTATION_TYPE.ANNOTATION,
+      event: 'user',
+      detector_index: 1,
+      partition_field_name: 'airline',
+      partition_field_value: 'AAL',
+    };
+  };
+
+  const testSetupAnnotations = testSetupJobConfigs.map((job) =>
+    createAnnotationRequestBody(job.job_id)
+  );
+
+  describe('update_annotations', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('ml/farequote');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+
+      // generate one annotation for each job
+      for (let i = 0; i < testSetupJobConfigs.length; i++) {
+        const job = testSetupJobConfigs[i];
+        const annotationToIndex = testSetupAnnotations[i];
+        await ml.api.createAnomalyDetectionJob(job);
+        await ml.api.indexAnnotation(annotationToIndex);
+      }
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+    });
+
+    it('should correctly update annotation by id', async () => {
+      const annotationsForJob = await ml.api.getAnnotations(jobIds[0]);
+      expect(annotationsForJob).to.have.length(1);
+
+      const originalAnnotation = annotationsForJob[0];
+      const annotationUpdateRequestBody = {
+        timestamp: Date.now(),
+        end_timestamp: Date.now(),
+        annotation: 'Updated annotation #0',
+        job_id: originalAnnotation._source.job_id,
+        type: ANNOTATION_TYPE.ANNOTATION,
+        event: 'model_change',
+        detector_index: 2,
+        partition_field_name: 'airline',
+        partition_field_value: 'ANA',
+        _id: originalAnnotation._id,
+      };
+      const { body } = await supertest
+        .put('/api/ml/annotations/index')
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(COMMON_REQUEST_HEADERS)
+        .send(annotationUpdateRequestBody)
+        .expect(200);
+
+      expect(body._id).to.eql(originalAnnotation._id);
+      expect(body.result).to.eql('updated');
+
+      const updatedAnnotation = await ml.api.getAnnotationById(originalAnnotation._id);
+      expect(updatedAnnotation?.annotation).to.eql(annotationUpdateRequestBody.annotation);
+      expect(updatedAnnotation?.detector_index).to.eql(annotationUpdateRequestBody.detector_index);
+      expect(updatedAnnotation?.event).to.eql(annotationUpdateRequestBody.event);
```

Comment:
Would be good to also validate the partition details.

Reply:
Updated in https://github.com/elastic/kibana/pull/73068/commits/f45a447e78d38637d6649fcb92e6d3235b13c906